### ISSUE TITLE
Use CatBoost MultiQuantile regression

### ISSUE THREAD

**Issue by swertz** [unit8co/darts#2967]
**Is your feature request related to a current problem? Please describe.**

CatBoost has the ability to regress multiple quantiles using a single model, with the `MultiQuantile` loss: https://catboost.ai/docs/en/concepts/loss-functions-regression#MultiQuantile

However, it seems that feature is not used by Darts: https://github.com/unit8co/darts/blob/master/darts/models/forecasting/catboost_model.py#L352

It might make the use of multiple quantiles much faster by avoiding the training of multiple models.

**Describe proposed solution**

It seems relatively straightforward to use `MultiQuantile` instead of `Quantile`. I'm happy to contribute a PR for this. The questions I would like to clarify are:
- how does this interact with multi-output regression, either for multiple components, or in the case of `multi_models=True` with `output_chunk_length>1`? As far as I understand, multi-regression is only supported for RMSE loss, correct?
- should this be left exposed as an option to the user?

**Comment by dennisbader** [comment]
Hi @swertz and thanks for raising this feature request. It's true that currently all SKLearn-like models that support likelihood="quantile" do so by fitting dedicated models per quantile.

We'd be very happy about a contribution :) 

Regarding your questions:
- We perform some checks [here](https://github.com/unit8co/darts/blob/12327c069e5db11fd15d6a26c573ee2ec523426e/darts/models/forecasting/sklearn_model.py#L1035) until line 1059 to determine:
  - is multi-output required? (e.g. either multivariate target and / or output_chunk_length>1 when multi_models=True)
    - if yes, does the model support multi-output natively? For multi quantile it would have to support multi-quantile for multiple columns
      - if yes, use the model as is
      - if no, wrap the model in our MultiOutputRegressor that fits a dedicated model for each target component and step in the output chunk 
- yes, we should expose this as a new likelihood option. We could call it "multiquantile" or similar.

How does that sound?

**Comment by swertz** [comment]
Thanks for the feedback @dennisbader !

> if yes, does the model support multi-output natively? For multi quantile it would have to support multi-quantile for multiple columns 

Indeed, as far as I understand this only works for using multiple quantiles for a single regressed target, so a multivariate target or output chunk would still need to be handled through the `MultiOutputRegressor`.

I'll try to have a look at this :) Would you mind giving a few pointers regarding where/how to add tests?

**Comment by dennisbader** [comment]
Hi @swertz, sure :) 
You can add dedicated CatBoost tests to `darts/tests/models/forecasting/test_sklearn_models.py`. There are already a couple of tests for multi-output available. But a dedicated test would make most sense to me.

One thing that we should also keep in mind is to make `SKLearnModel.get_estimator()` work with the new multi-quantile case.

And on a last note, if multi-quantile is also natively available for other models (LightGBM, XGBoost, ...) then we could add support for that as well :)
